shell/initialize/client.js

```diff
@@ -256,6 +278,8 @@ function resolveComponents(route) {
   });
 }
 
+// this looks like support for Nuxt middleware's (middleware prop on a component)
+// couldn't we replace this functionality (especially after removing Nuxt layouts) to something like https://markus.oberlehner.net/blog/implementing-a-simple-middleware-with-vue-router/?
```

Comment:
We could. I'd probably prefer to just remove middleware entirely if it's possible.